tests/test_game_time.py

```diff
@@ -60,3 +61,14 @@ def ow_player_data_response_not_found():
     ctx = await pg.prepare_game_times_context([OW_ID])
     result = await pg.get_game_time(OW_ID, ctx)
     assert result == GameTime(OW_ID, minutes, None)
+
+async def test_last_played_time(pg, config_data):
+    pg.local_client.config_parser = ConfigParser(config_data)
+
+    ctx = await pg.prepare_game_times_context(["21298"])
+    result = await pg.get_game_time("21298", ctx)
+    assert result == GameTime("21298", None, None)
+
+    ctx = await pg.prepare_game_times_context(["1214607983"])
+    result = await pg.get_game_time("1214607983", ctx)
+    assert result == GameTime("1214607983", None, 1441712029)
```

Comment:
This is fine (though not "perfect")

Best unit-test tests only one thing and does not rely on implementation of other things like ConfigParser which should be tested independently.
But this case is acceptable, because ConfigParser is simple class that does not makes any side-effects like reading files and so on. In case of more complicated things, all of those should be mocked for example:

```python

# conftest

@pytest.fixture()
def config_parser():
    return MagicMock()


@pytest.fixture()
def local_client_mock(config_parser):
    mock = Mock(spec=LocalClient)
    mock.launch_game = AsyncMock()
    mock.config_parser = config_parser
    return mock
```
```python
# test file

from src.definitions import Blizzard, ConfigGameInfo


@pytest.mark.asyncio
async def test_last_played_when_unknown_game_in_config(pg, config_parser):
    game = Blizzard['diablo3']
    unknown_game_uid = 'unknown_game'
    with pytest.raises(KeyError):
        assert Blizzard[unknown_game_uid]  # test precondition

    config_parser.games = [
        ConfigGameInfo(unknown_game_uid, 'mock', '111111111'),
        ConfigGameInfo(game.uid, 'diablo3_enus', '1441712029')
    ]

    ctx = await pg.prepare_game_times_context([game.blizzard_id])
    result = await pg.get_game_time(game.blizzard_id, ctx)
    assert result.game_id == game.blizzard_id
    assert result.last_played_time == 1441712029
```
Test from this example is failing when keeping `break` from previous discussion and is fixed when giving `continue`

But as I said - its OK to have explicite `ConfigParser(mock_data)` here, you don't have to change it

Reply:
@bartok765 thanks your advice , i should changed that,  thats imporved the test case.